engine/src/main/resources/org/camunda/bpm/engine/impl/mapping/entity/HistoricProcessInstance.xml

```diff
@@ -626,6 +636,21 @@
               )
             </if>
 
+            <if test="query.activityIds != null and query.activityIds.length > 0">
```

Comment:
❓ Do we need to consider adding some indexes here? To ensure performance for the modification. Or the existing indexes will be good enough?

Reply:
Let's be careful with indices on history tables. I would only add them when we get feedback. Adding a new index for existing databases with tables that have huge amounts of data is a costly operation, and it slows down process execution since history insertion is slower.